lib/phoenix_live_view.ex

```diff
@@ -1134,6 +1135,52 @@ defmodule Phoenix.LiveView do
       class will be applied in conjunction with `"phx-disconnected"` if connection
       to the server is lost.
 
+  All `phx-` event bindings apply their own css classes when pushed. For example
+  the following markup:
+
+      <button phx-click="clicked" phx-window-keydown="key">...</button>
+
+  In the case of forms, when a `phx-change` is sent to the server, the input element
+  which emitted the change receives the `phx-change-loading` class, along wiht the
+  parent form tag.
+
+  On click, would receive the `phx-click-loading` class, and on keydown would receive
+  the `phx-keydown-loading` class. The css loading classes are maintained until an
+  acknowledgement is received on the client for the pushed event. The following events
+  receive css loadng classes:
+
+    - `phx-click` - `phx-click-loading`
+    - `phx-change` - `phx-change-loading`
+    - `phx-submit` - `phx-submit-loading`
+    - `phx-focus` - `phx-focus-loading`
+    - `phx-blur` - `phx-blur-loading`
+    - `phx-window-keydown` - `phx-keydown-loading`
+    - `phx-window-keyup` - `phx-keyup-loading`
+
+  For live page navigation via `live_redirect` and `live_patch`, as well as form
+  submits via `phx-submit`, the JavaScript events `"phx:page-loading-start"` and
+  `"phx:page-loading-stop"` are dispatched on window. Additionally, any `phx-`
+  event may dispatch page loading events by annotating the DOM element with
+  `phx-page-loading`. This is useful for showing main page loading status, for example:
+
+      // app.js
+      import NProgress from "nprogress"
+      window.addEventListener("phx:page-loading-start", info => NProgress.start())
+      window.addEventListener("phx:page-loading-stop", info => NProgress.done())
+
+  The `info` object will contain a `kind` key, with values one of:
+
+    - `"redirect"` - the event was triggered by a redirect
+    - `"patch"` - the event was triggered by a patch
+    - `"initial"` - the event was triggered by initial page load
+    - `"element"` - the event was triggered by a `phx-` bound element, such as `phx-click`
```

Comment:
We are missing "submit" from this list or does it fall under "element"?

Reply:
falls under element, but we can add another kind?